File: muxer-commons/src/main/java/pt/um/tf/taskmuxer/commons/error/UnknownClientException.java
package pt.um.tf.taskmuxer.commons.error;

import io.atomix.catalyst.buffer.BufferInput;
import io.atomix.catalyst.buffer.BufferOutput;
import io.atomix.catalyst.serializer.CatalystSerializable;
import io.atomix.catalyst.serializer.Serializer;

public class UnknownClientException extends Exception implements CatalystSerializable {
    @Override
    public String getMessage() {
        return super.getMessage() + System.lineSeparator() + "Unknown Client";
    }

    @Override
    public void writeObject(BufferOutput<?> buffer, Serializer serializer) {}

    @Override
    public void readObject(BufferInput<?> buffer, Serializer serializer) {}
}

File: muxer-commons/src/main/kotlin/pt/um/tf/taskmux/commons/error/UnknownClientException.kt
package pt.um.tf.taskmux.commons.error

import io.atomix.catalyst.buffer.BufferInput
import io.atomix.catalyst.buffer.BufferOutput
import io.atomix.catalyst.serializer.CatalystSerializable
import io.atomix.catalyst.serializer.Serializer

class UnknownClientException : Exception(), CatalystSerializable {
    override val message: String?
        get() = "Unknown Client"

    override fun writeObject(buffer: BufferOutput<*>?, serializer: Serializer?) {}

    override fun readObject(buffer: BufferInput<*>?, serializer: Serializer?) {}

}
